components/suggest/src/db.rs

```diff
@@ -128,63 +128,72 @@ impl<'a> SuggestDao<'a> {
 
     /// Fetches suggestions that match the given query from the database.
     pub fn fetch_suggestions(&self, query: &SuggestionQuery) -> Result<Vec<Suggestion>> {
-        if let Some(suggestion) = self.fetch_yelp_suggestion(query)? {
-            return Ok(vec![suggestion]);
-        }
-
-        let keyword_lowercased = &query.keyword.to_lowercase();
-        let (keyword_prefix, keyword_suffix) = split_keyword(keyword_lowercased);
         let suggestions_limit = query.limit.unwrap_or(-1);
+        let suggestions =
+            query
+                .providers
+                .clone()
+                .into_iter()
+                .try_fold(vec![], |mut acc, provider| {
```

Comment:
Nice use of `try_fold`! 🏆 Those nested `Result<Vec<Suggestion>>` really make working with them awkward, but this looks great.

There's a Rust crate inspired by Python's called [`itertools`](https://docs.rs/itertools/latest/itertools/), and it has a [`flatten_ok`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.flatten_ok) helper for this exact case. It looks like `nimbus-fml` and `glean` already use `itertools`, so we could use it here if we wanted—but it doesn't seem worth it just for one method, when your approach is already clean and readable.